File: leetcode_2665/yoonalc.ts
// Runtime 81ms Beats 12.01%
// Memory 45.6MB Beats 42.76%

type ReturnObj = {
    increment: () => number,
    decrement: () => number,
    reset: () => number,
}

export function createCounter(init: number): ReturnObj {
    let count = init
    return {
        increment: () => count+=1,
        decrement: () => count-=1,
        reset: () => count=init,
    }
};

File: leetcode_442/yoonalc.go
// Runtime: 50 ms Beats 78.43%
// Memory: 7.4 MB Beats 78.43%
package find_all_duplicates_in_an_array

func findDuplicates(nums []int) []int {
	var count []int

	abs := func(x int) int {
		if x > 0 {
			return x
		}
		return -x
	}

	for i := range nums {
		idx := abs(nums[i]) - 1
		if nums[idx] > 0 {
			nums[idx] *= -1
		} else {
			count = append(count, abs(nums[i]))
		}
	}

	return count
}
